testsuite/HOWTO.md

```diff
@@ -0,0 +1,176 @@
+============ 5497 passed, 31 skipped, 1 xfailed in 1056.77 seconds =============
+```
+
+### Coverage ###
+We test code coverage of the unit tests with the  [coverage](http://nedbatchelder.com/code/modules/rees-coverage.html) plugin of pytest. Currently, this is done automatically as part of the Travis job and is viewable on **[coveralls](https://coveralls.io/r/MDAnalysis/mdanalysis?branch=develop)**.
+
+If you want to generate a coverage report manually, you will need `pytest-cov` installed.
+`pip install pytest-cov`
+
+Then you can run -
+```
+cd testsuite
+pytest --numprocesses 4 --cov=MDAnalysis --cov-report html
+```
+
+# Details #
+We are borrowing some of NumPy's testing frame work; thus, numpy **must** be installed for the tests to run at all.
+
+## Running tests from within python ##
+Run all the tests with
+```
+python -c 'from MDAnalysisTests import run; run()'
+```
+
+## Running tests from the command line ##
+Instead of running tests from within python, one can also run them via the command line.
+Go into the tests directory (or the package root)
+```
+cd /testsuite/MDAnalysisTests
+```
+and invoke pytest directly to run **all tests** on four processors in parallel ("`%`" is the shell prompt and should _not_ be typed):
+```
+% pytest --numprocesses 4
+```
+Again, to run the tests in parallel (use `--numprocesses 4`) you need to install `pytest-xdist` via `pip install pytest-xdist`
+
+When you have written a **new unit test** it is helpful to check that it passes without running the entire suite. For example, in order to test everything in, say, `test\_selections.py` run
+```
+% pytest test_selections
+..............
+----------------------------------------------------------------------
+Ran 14 tests in 3.421s
+
+OK
+```
+One can also test individual test classes. For instance, after working on the XYZReader one can check just the TestCompressedXYZReader tests with
+```
+% pytest test_coordinates::TestCompressedXYZReader
+....
+----------------------------------------------------------------------
+Ran 4 tests in 0.486s
+
+OK
+```
+where we are testing the class `TestCompressedXYZReader` which can be found in the module (file) `test\_coordinates.py`.
+
```

Comment:
@kain88-de , I just want to add that @utkbansal is starting from a document that was pretty messy and repetitious to begin with so I think we should really encourage @utkbansal to be bold: he should feel that he can remove and condense sections when it leads to a clearer document.